docs/client/session.md

```diff
@@ -0,0 +1,25 @@
+# Semantic conventions for session
+This document defines semantic conventions to apply to client-side applications when tracking sessions.
+
+A session is collection of Logs, Events, and Spans associated with a specific device or user within a specific period of time. 
+A session is represented by a unique identifier that is attached to Logs, Events, and Spans as an attribute.
```

Comment:
Should we say that the session.id value is generated by the SDK? I want to make sure that it is not misunderstood as a session supplied by the application (e.g. authenticated user session).

Reply:
I suggest defining Session as follows:

> Session is defined as a single run of an application by an end user, encompassing all the activities performed by the application and the actions executed by the end user. It has a specific duration and is usually concluded when there is no activity, or it can be explicitly ended after a certain maximum duration.
> 
> Consequently, a Session is represented as a collection of Logs, Events, and Spans emitted by the Client Application throughout the Session's duration. Each Session is assigned a unique identifier, which is included as an attribute in the Logs, Events, and Spans generated during the Session's lifecycle.

Reply:
Not sure if we should be that specific in the spec, as the sessionId can be provided by the application if it already has a notion of session, or the SDK if it doesn't or wants to defer the session lifecycle to the SDK.

Reply:
@martinkuba I think that's reasonable. The term 'user' in the definition could be misconstrued as an authenticated user, as opposed the general "person interacting with the interface". Do you think your suggestion is enough to waylay this confusion? 

@scheler I think that's a pretty good definition, but I don't think the "single run of an application" is necessarily accurate. I imagine an application may crash and be immediately resumed by the user, and still be one session. I guess it depends on how we define "single run"?  I think "a collection of [signals] associated with a single device or user over a span of time" is more precise wording.  Maybe reword it as:
> Session is defined as the period of time encompassing all activities performed by the application and the actions executed by the end user. The period of time is defined by the initial activity and concluded after a period of inactivity, or explicitly ended after a maximum duration is reached.  
>
> Consequently, a Session is represented as a collection of Logs, Events, and Spans emitted by the Client Application throughout the Session's duration. Each Session is assigned a unique identifier, which is included as an attribute in the Logs, Events, and Spans generated during the Session's lifecycle.

Reply:
> The period of time is defined by the initial activity and concluded after a period of inactivity, or explicitly ended after a maximum duration is reached.

I think even this is too prescriptive. I think removing it but including everything else would be precise enough to be useful

Reply:
> I think that's a pretty good definition

I took the help of ChatGPT to frame the sentences :-)

Agree on both the suggestions - removing 'single run', and also not prescribing how a session ends. So from the revised text above, we could remove "The period of time is defined by the initial activity and concluded after a period of inactivity, or explicitly ended after a maximum duration is reached.".

Reply:
> Should we say that the session.id value is generated by the SDK? I want to make sure that it is not misunderstood as a session supplied by the application (e.g. authenticated user session).

+1, there should be a clear statement that the session ID must not be re-used from the application, because this can lead to security issues.